JYH/3Day/문자열 다루기.js

```diff
@@ -0,0 +1,12 @@
+function solution(s) {
+    if (s.length === 4 || s.length === 6) {
+        if(s.split('').every(x => !isNaN(x)))
+        {
+            return true;
+        } else {
+            return false;
+        }
+    } else {
+        return false;
+    }
+}
```

Comment:
```suggestion
function solution(s) {
    if (s.length === 4 || s.length === 6) {
        const isNumericString = s.split('').every(x => !isNaN(x));
        
        if (isNumericString) {
            return true;
        } else {
            return false;
        }
    } else {
        return false;
    }
}
```
@jungyoonho128 
`s.split('').every(x => !isNaN(x))` 와 같은 로직은 변수로 분리해주세요.
if문 내 조건식이 복잡하다면 가독성이 매우 떨어집니다. 또한, 제3자가 봤을 때 어떤 의미를 하는지 파악이 힘들어요.

좋은 코드가 무엇일까요? 고민해보신적 있으실까요? ㅎㅎㅎ

Reply:
@jungyoonho128 제안드린 코드도 사실 단계적으로 더 개선해 볼 수 있어요.
```ts
function solution(s) {
    if (s.length === 4 || s.length === 6) {
        const isNumericString = s.split('').every(x => !isNaN(x));
        
        return isNumericString
    } else {
        return false;
    }
}
```
1. isNumericString 변수를 통해 하단 if ... else문은 불 필요해집니다.

```ts
function solution(s) {
    if (s.length === 4 || s.length === 6) {
        const isNumericString = s.split('').every(x => !isNaN(x));
        
        return isNumericString
    }
    return false;
}
```
2. if ~ else문에서 return이 있다면 else 문은 제거해도 됩니다. 어차피 if문의 조건식에 걸려서 return이 되고 함수가 종료 될 것이고, 그렇지 않다면 다음 return문에 걸릴 것이기 때문에 else문은 필요없어져요

Reply:
좋은 코드는 얼마나 간결하게 코드를 짠 사람이 원하는게 무엇인지 전할 수 있는 코드라고 생각합니다. 이렇게 생각해보면 저는 안좋은 코드를 짜고 있단 생각이 드네요.. 알려주신 것 참고하면서 제 스스로 더 찾아보고 열심히 공부하도록 하겠습니다 알려주셔서 감사합니다!

Reply:
@jungyoonho128 좋은 코드라는게 사실 광범위적이긴 해요. 👍 꼭 간단하고, 간결할 필요는 없지만, 가독성 좋고 목적과 의도를 제3자에게 온전히 전달하는 코드도 좋은 코드중 하나라고 볼 수 있을 것 같습니다